.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '31 4 * * 0'
```

Comment:
do we need this?

Reply:
This is the standard from codeql with a cron job. I used that. If you think we should remove it let me know I can do that.

Reply:
I might be overlooking something but I see no need to have a cron

Reply:
The reason for having the cron is that if for example if there hasn't been a code change for "X" number of days and the codeql team figures out a new vulnerability pattern which the codeql team updates its codebase.

The new codeql codebase would be run against the source code which can identify Vulns when there is a cronjob. If there isn't this check then only when the next PR comes in that's when an issue would be identified.